### ISSUE TITLE
[ecbuff] suggestion : #error if neither ECB_WRITE_DROP or _OVERWRITE specified

### ISSUE THREAD

**Issue by fenugrec** [EliasOenal/etools#3]
As I understand, specifying neither of ECB_WRITE_DROP or ECB_WRITE_OVERWRITE, while ECB_EXTRA_CHECKS is undefined,  may break ecbuff, or at the very least behave strangely ?

I suggest adding some ifdef magic to abort if none of those two are defined.
Maybe something like this ?
```
#if !(defined(ECB_WRITE_OVERWRITE) ^ defined(ECB_WRITE_DROP))
#error Exactly one of ECB_WRITE_OVERWRITE or ECB_WRITE_DROP must be defined.
#endif
```
(bonus, it also checks if the two were specified !)

**Comment by EliasOenal** [comment]
I agree that defining neither usually isn't a good choice. The reason I left it as an option was for performance considerations. It can be useful in situations where one has to crank out a little bit of additional throughput from a microcontroller. So while I don't think it should be removed altogether, hiding it behind a define could make this a bit more explicit. Alternatively maybe just explaining this a bit better would be enough, currently it is only passingly mentioned in the description of ECB_EXTRA_CHECKS. What would you suggest?

**Comment by fenugrec** [comment]
> The reason I left it as an option was for performance considerations. [...]

Fair enough. In that case, as you said, just a short explanation somewhere in the headers near those defines would be a nice addition IMO.
